test/integration.test.ts

```diff
@@ -0,0 +1,183 @@
+    Context,
+    describe,
+    it,
+    Message,
+    spy,
+    Update,
+    User,
+} from "./deps.test.ts";
+
+const bot = new Bot("dummy_token");
+
+const getDummyUpdate = ({ userInput, language, noChat }: {
+    userInput?: string;
+    language?: string;
+    noChat?: boolean;
+} = {}) => {
+    const u = { id: 42, first_name: "yo", language_code: language } as User;
+    const c = { id: 100, type: "private" } as Chat;
+    const m = {
+        text: userInput,
+        from: u,
+        chat: noChat ? undefined : c,
+    } as Message;
+    const update = {
+        message: m,
+    } as Update;
+
+    return update;
+};
+
+describe("Integration", () => {
+    describe("setCommands", () => {
+        it("should call setMyCommands with valid commands", async () => {
+            const myCommands = new CommandGroup();
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+
+            await myCommands.setCommands({
+                api: {
+                    raw: { setMyCommands: setMyCommandsSpy },
+                } as unknown as Api,
+            });
+
+            assertSpyCalls(setMyCommandsSpy, 1);
+            assertSpyCall(setMyCommandsSpy, 0, {
+                args: [{
+                    commands: [{ command: "command", description: "_" }],
+                    language_code: undefined,
+                    scope: {
+                        type: "default",
+                    },
+                }],
+            });
+        });
+
+        it("should error when commands have custom prefixes", async () => {
+            const myCommands = new CommandGroup({ prefix: "!" });
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+
+            await assertRejects(() =>
+                myCommands.setCommands({
+                    api: {
+                        raw: { setMyCommands: setMyCommandsSpy },
+                    } as unknown as Api,
+                })
+            );
+
+            assertSpyCalls(setMyCommandsSpy, 0);
+        });
+    });
+
+    describe("ctx.setMyCommands", () => {
+        it("should call setMyCommands with valid commands", async () => {
+            const myCommands = new CommandGroup();
+            myCommands.command("command", "_", (_, next) => next());
+
+            const setMyCommandsSpy = spy(resolvesNext([true] as const));
+            const bot = new Bot<Context & CommandsFlavor>("dummy_token", {
+                botInfo: {
+                    id: 1,
+                    is_bot: true,
+                    username: "",
+                    can_join_groups: true,
+                    can_read_all_group_messages: true,
+                    supports_inline_queries: true,
+                    first_name: "",
+                    can_connect_to_business: true,
+                    has_main_web_app: false,
```

Comment:
ts compling:
`Object literal may only specify known properties, and 'has_main_web_app' does not exist in type 'UserFromGetMe'.deno-ts(2353)`
`has_main_web_app` is not used in the test's anyways

Reply:
I was getting this error too, but I can't reproduce it now, neither does the CI seem to run into it
Could you try `deno task check --reload` to see if it still persists, please?

Reply:
```
TS2345 [ERROR]: Argument of type 'import("https://deno.land/x/grammy_types@v3.8.0/manage.ts").UserFromGetMe' is not assignable to parameter of type 'import("https://deno.land/x/grammy_types@v3.13.0/manage.ts").UserFromGetMe'.
                const ctx = new Context(update, api, me);
                                                     ~~
    at file/.../command.test.ts:74:54
    
    repeat x20
    ```

Reply:
Currently `deno test` throws multiple errores, even after `deno task check --reload` and `deno cache $FILE`.
```
TS2345 [ERROR]: Argument of type 'import("https://deno.land/x/grammy_types@v3.8.0/manage.ts").UserFromGetMe' is not assignable to parameter of type 'import("https://deno.land/x/grammy_types@v3.13.0/manage.ts").UserFromGetMe'.
                        const ctx = new Context(update, api, me);
                                                             ~~
    at file:///.../test/command.test.ts:516:62

TS2345 [ERROR]: Argument of type 'import("https://deno.land/x/grammy_types@v3.8.0/manage.ts").UserFromGetMe' is not assignable to parameter of type 'import("https://deno.land/x/grammy_types@v3.13.0/manage.ts").UserFromGetMe'.
  Property 'has_main_web_app' is missing in type 'import("https://deno.land/x/grammy_types@v3.8.0/manage.ts").UserFromGetMe' but required in type 'import("https://deno.land/x/grammy_types@v3.13.0/manage.ts").UserFromGetMe'.
    const ctx = new Context(update, api, me) as CommandsFlavor<Context>;
                                         ~~
    at file:///.../test/context.test.ts:73:42

    'has_main_web_app' is declared here.
      has_main_web_app: boolean;
      ~~~~~~~~~~~~~~~~
        at https://deno.land/x/grammy_types@v3.13.0/manage.ts:65:3

Found 20 errors.

```

Reply:
that's weird, I ran it multiple times locally, and CI is passing too 🤔